idd/Energy+.idd.in

```diff
@@ -7604,6 +7604,54 @@ MaterialProperty:VariableThermalConductivity,
        \units W/m-K
        \type real
 
+MaterialProperty:VariableThermalAbsorptance,
```

Comment:
General question. Are `MaterialProperty` orthogonal? Can a `Material` have any combination of them?

Reply:
I believe for MaterialProperty:x, there can only be at most one of each kind. There can be MaterialProperty:a, MaterialProperty:b, MaterialProperty:c, etc. for the same material where a, b, c are different add-ons. For example, the `MaterialProperty:PhaseChange` object has just one "Name" field pointing to the material object it references. There is not a name for the add-on itself. So one material cannot have multiple `MaterialProperty:PhaseChange` objects as they will not be distinguishable. 

The two I added do have their own names in addition to the material they reference, but I still only allow for one of each kind for each material as other add-ons.

Reply:
That part I get. I meant are the different `MaterialProperty` objects orthogonal to each other relative to a single `Material` object. Can a single `Material` have `MaterialProperty:PhaseChange` and `MaterialProperty:VariableThermalAbsorptance` and `MaterialProperty:MoisturePenetrationDepth` and the rest of them?

Reply:
These are the MaterialProperty: ones. From the names it looks like they should be orthogonal (although need to double check the PhaseChange and PhaseChangeHysterisis)
<img width="437" alt="image" src="https://user-images.githubusercontent.com/13421161/215599581-9849701c-3c7d-4cef-98bb-c52dca88ddd2.png">

Reply:
Some of these are mutually exclusive for any given surface. It's actually kind of messy. The two new ones, however can work in combination with any of the others.

MaterialProperty:MoisturePenetrationDepth:Settings - requires HeatBalanceAlgorithm=MoisturePenetrationDepthConductionTransferFunction

MaterialProperty:PhaseChange and MaterialProperty:PhaseChangeHysteresis require HeatBalanceAlgorithm = CondFD (ConductionFiniteDifference), but you'd only use one or the other.

MaterialProperty:VariableThermalConductivity requires HeatBalanceAlgorithm = CondFD Can be used together with one of MaterialProperty:PhaseChange*

MaterialProperty:HeatAndMoistureTransfer:* are user as a group and require HeatBalanceAlgorithm = CombinedHeatAndMoistureFiniteElement

MaterialProperty:GlazingSpectralData only applies to windows, all the others only apply to opaque surfaces.

So, not very orthogonal, kind of skewed or twisted.

Reply:
Thanks for the clarification @mjwitte. Glad the new ones can work together with the rest add-ons.

Reply:
Oh what a tangled web we weave ...